core/src/test/java/org/elasticsearch/index/query/PercolatorQueryBuilderTests.java

```diff
@@ -75,6 +78,18 @@ private PercolatorQueryBuilder doCreateTestQueryBuilder(boolean indexedDocument)
         }
     }
 
+    /**
+     * prevent fields in the "document" field from being shuffled randomly, because it later is parsed to
+     * a {@link BytesReference} and even though the documents are the same, equals will fail when comparing
+     * BytesReference
+     */
+    @Override
+    protected Set<String> provideShuffleproofFields() {
```

Comment:
the ones returned here are the fields that can be shuffled or that should not be shuffled? Not sure if the name is consistent with the behaviour